### ISSUE TITLE
"undefined reference to `kdu_supp::" while compiling Gdal with kakadu 8.3 support

### ISSUE THREAD

**Issue by MathewNWSH** [OSGeo/gdal#8417]
## Expected behavior and actual behavior.
In reference to https://github.com/OSGeo/gdal/issues/4306, I would like to report that when compiling GDAL independently using the official Dockerfile and based on the Ubuntu image with the Kakadu library installed, I am still encountering an error "undefined reference to `kdu_supp::":
![image](https://github.com/OSGeo/gdal/assets/106817115/8b60fe41-c300-43bf-8e8a-3586f01ce819)
Full log file -> [gdal_log.log](https://github.com/OSGeo/gdal/files/12648545/gdal_log.log)

## Steps to reproduce the problem.

1.Create an Ubuntu image with Kakadu installed. During installation, I based it on https://trac.osgeo.org/gdal/wiki/JP2KAK. Below is an example Dockerfile.
```
# Use the official Ubuntu 20.04 as the base image

ARG BASE_UBUNTU=ubuntu:22.04
FROM $BASE_UBUNTU as builder

# Set environment variables if needed
# ENV MY_VARIABLE=value

RUN mkdir -p /kakadu

# Update the package lists and install necessary packages
RUN apt-get update && \
    apt-get install -y wget unzip make g++

# Add your application code to the container
COPY v8_3-02164E/ /kakadu/



RUN cd kakadu/coresys/make && \
    make -f Makefile-Linux-x86-64-gcc && \
    cd ../..
    
ENV CXXFLAGS -fPIC

RUN cd /kakadu/apps/make && \
    make -f Makefile-Linux-x86-64-gcc

ENV kdu_dir /kakadu/lib/Linux-x86-64-gcc/
# Set the working directory
# WORKDIR /app

# Specify the command to run your application
# CMD [ "command", "arg1", "arg2" ]
```

2.Then, in the Dockerfile from https://github.com/OSGeo/gdal/tree/master/docker/ubuntu-small, add lines 182-185, the way below (DKDU options):
```
# Build GDAL
ARG GDAL_VERSION=master
ARG GDAL_RELEASE_DATE
ARG GDAL_BUILD_IS_RELEASE
ARG GDAL_REPOSITORY=OSGeo/gdal

RUN . /buildscripts/bh-set-envvars.sh \
    && if test "${GDAL_VERSION}" = "master"; then \
        export GDAL_VERSION=$(curl -Ls https://api.github.com/repos/${GDAL_REPOSITORY}/commits/HEAD -H "Accept: application/vnd.github.VERSION.sha"); \
        export GDAL_RELEASE_DATE=$(date "+%Y%m%d"); \
    fi \
    && if test "x${GDAL_BUILD_IS_RELEASE:-}" = "x"; then \
        export GDAL_SHA1SUM=${GDAL_VERSION}; \
    fi \
    && mkdir gdal \
    && wget -q https://github.com/${GDAL_REPOSITORY}/archive/${GDAL_VERSION}.tar.gz -O - \
        | tar xz -C gdal --strip-components=1 \
    && cd gdal \
    && if test "${RSYNC_REMOTE:-}" != ""; then \
        echo "Downloading cache..."; \
        rsync -ra ${RSYNC_REMOTE}/gdal/${GCC_ARCH}/ $HOME/; \
        echo "Finished"; \
        # Little trick to avoid issues with Python bindings
        printf "#!/bin/sh\nccache %s-linux-gnu-gcc \$*" "${GCC_ARCH}" > ccache_gcc.sh; \
        chmod +x ccache_gcc.sh; \
        printf "#!/bin/sh\nccache %s-linux-gnu-g++ \$*" "${GCC_ARCH}" > ccache_g++.sh; \
        chmod +x ccache_g++.sh; \
        export CC=$PWD/ccache_gcc.sh; \
        export CXX=$PWD/ccache_g++.sh; \
        ccache -M 1G; \
    fi \
    && mkdir build \
    && cd build \
    && CFLAGS='-DPROJ_RENAME_SYMBOLS -O2' CXXFLAGS='-DPROJ_RENAME_SYMBOLS -DPROJ_INTERNAL_CPP_NAMESPACE -O2' \
       cmake .. \
        -DCMAKE_INSTALL_PREFIX=/usr \
        -DPROJ_INCLUDE_DIR="/build${PROJ_INSTALL_PREFIX-/usr/local}/include" \
        -DPROJ_LIBRARY="/build${PROJ_INSTALL_PREFIX-/usr/local}/lib/libinternalproj.so" \
        -DGDAL_USE_TIFF_INTERNAL=ON \
        -DGDAL_USE_GEOTIFF_INTERNAL=ON \
        -DBUILD_TESTING=OFF \
        -DKDU_ROOT=/kakadu/ \
        -DKDU_INCLUDE_DIR=/kakadu/ \
        -DKDU_LIBRARY=/kakadu/lib/Linux-x86-64-gcc/libkdu_v83R.so \
        -DKDU_AUX_LIBRARY=/kakadu/lib/Linux-x86-64-gcc/libkdu_v83R.so \
    && make -j$(nproc) \
    && make install DESTDIR="/build" \
    && cd .. \
    && if test "${RSYNC_REMOTE:-}" != ""; then \
        ccache -s; \
        echo "Uploading cache..."; \
        rsync -ra --delete $HOME/.ccache ${RSYNC_REMOTE}/gdal/${GCC_ARCH}/; \
        echo "Finished"; \
        rm -rf $HOME/.ccache; \
        unset CC; \
        unset CXX; \
    fi \
    && cd .. \
    && rm -rf gdal \
    && mkdir -p /build_gdal_python/usr/lib \
    && mkdir -p /build_gdal_python/usr/bin \
    && mkdir -p /build_gdal_version_changing/usr/include \
    && mv /build/usr/lib/python*            /build_gdal_python/usr/lib \
    && mv /build/usr/lib                    /build_gdal_version_changing/usr \
    && mv /build/usr/include/gdal_version.h /build_gdal_version_changing/usr/include \
    && mv /build/usr/bin/*.py               /build_gdal_python/usr/bin \
    && mv /build/usr/bin                    /build_gdal_version_changing/usr \
    && for i in /build_gdal_version_changing/usr/lib/${GCC_ARCH}-linux-gnu/*; do ${GCC_ARCH}-linux-gnu-strip -s $i 2>/dev/null || /bin/true; done \
    && for i in /build_gdal_python/usr/lib/python3/dist-packages/osgeo/*.so; do ${GCC_ARCH}-linux-gnu-strip -s $i 2>/dev/null || /bin/true; done \
    && for i in /build_gdal_version_changing/usr/bin/*; do ${GCC_ARCH}-linux-gnu-strip -s $i 2>/dev/null || /bin/true; done
```
## Operating system
ubuntu:22.04

## GDAL version and provenance
OSGeo/gdal:master (18.09.2023)

**Comment by rouault** [comment]
>        -DKDU_AUX_LIBRARY=/kakadu/lib/Linux-x86-64-gcc/libkdu_v83R.so \

This should be (a for auxiliary):

        -DKDU_AUX_LIBRARY=/kakadu/lib/Linux-x86-64-gcc/libkdu_a83R.so \

**Comment by MathewNWSH** [comment]
@rouault Yup, but there is no libkdu_a83R.so file in my kakadu installation directory (only libkdu_v83R.so exists).
![image](https://github.com/OSGeo/gdal/assets/106817115/a2e8531b-6b48-4d36-bbb7-3256044451a7)

**Comment by rouault** [comment]
> @rouault Yup, but there is no libkdu_a83R.so file in my kakadu installation directory (only libkdu_v83R.so exists). ![image](https://user-images.githubusercontent.com/106817115/268669746-a2e8531b-6b48-4d36-bbb7-3256044451a7.png)

you also need to run "make" from the managed/make/ subdirectory of your kakadu sources

**Comment by MathewNWSH** [comment]
> > @rouault Yup, but there is no libkdu_a83R.so file in my kakadu installation directory (only libkdu_v83R.so exists). ![image](https://user-images.githubusercontent.com/106817115/268669746-a2e8531b-6b48-4d36-bbb7-3256044451a7.png)
> 
> you also need to run "make" from the managed/make/ subdirectory of your kakadu sources

Yes, that' true here is updated version of the dockerfile for properly configured kakadu:
```
# Use the official Ubuntu 20.04 as the base image

ARG BASE_UBUNTU=ubuntu:22.04
FROM $BASE_UBUNTU as builder

# Set environment variables if needed
# ENV MY_VARIABLE=value

RUN mkdir -p /kakadu

# Update the package lists and install necessary packages
RUN apt-get update && \
    apt-get install -y wget unzip make g++ openjdk-11-jdk

# Add your application code to the container
COPY v8_3-02164E/ /kakadu/



RUN cd kakadu/coresys/make && \
    make -f Makefile-Linux-x86-64-gcc && \
    cd ../..
    
ENV CXXFLAGS -fPIC

#RUN cd /kakadu/managed/make/ && \
#    make -f Makefile-Linux-x86-64-gcc
    
RUN cd /kakadu/apps/make && \
    make -f Makefile-Linux-x86-64-gcc

ENV kdu_dir /kakadu/lib/Linux-x86-64-gcc/
# Set the working directory
# WORKDIR /app

# Specify the command to run your application
# CMD [ "command", "arg1", "arg2" ]

```
I have one more question. In relation to the https://github.com/OSGeo/gdal/issues/8275#issuecomment-1693576221 namely:
How to use the JPIPKAK driver instead of JP2KAK for example in this command
```
time gdalwarp -t_srs 'EPSG:3857' -q -of JPEG --config CPL_VSIL_CURL_CHUNK_SIZE 10485760 --config GDAL_NUM_THREADS ALL_CPUS --config CPL_DEBUG ON -te 3663787.0598015636 6899924.7342886925 3665141.729177569 6901279.403664698 /vsicurl/https://s3.waw3-1.cloudferro.com/swift/v1/CXD/T36UVD_20230701T085601_TCI_10m.jp2 dckr2.jp2
```
Thanks

**Comment by jratike80** [comment]
If you have Kakadu sources then you have also the "kdu_server" demo application. The usage_examples file https://[kakadusoftware.com/wp-content/uploads/Usage_Examples.txt](https://kakadusoftware.com/wp-content/uploads/Usage_Examples.txt) gives examples about how to start the server and also about how to use it with the kdu_show demo application. Unfortunately I could not make kdu_server to run properly on my Windows machine when I tried it a few minutes ago. The server stopped immediately when I asked kdu_show to read a remote .jp2 file from the server. I think I will not spend much time for investigating that.
I used to host kdu_server years ago when people still had dial-up modems. I remember that it felt fast, but maybe the feeling today with 4.8 kbit/sec would be different. I stopped the hosting because nobody really had JPIP clients. The situation is about the same today.
